packages/sso/src/routes/sso.ts

```diff
@@ -628,6 +631,17 @@ export const registerSSOProvider = <O extends SSOOptions>(options: O) => {
 					message: "Invalid issuer. Must be a valid URL",
 				});
 			}
+
+			if (body.samlConfig?.idpMetadata?.metadata) {
+				const maxMetadataSize =
+					options?.saml?.maxMetadataSize ?? DEFAULT_MAX_SAML_METADATA_SIZE;
+				if (body.samlConfig.idpMetadata.metadata.length > maxMetadataSize) {
+					throw new APIError("BAD_REQUEST", {
+						message: `IdP metadata exceeds maximum allowed size (${maxMetadataSize} bytes)`,
```

Comment:
This is not comparing bytes size, use byteLength with buffer or something

Reply:
Thank you for this... I followed the pattern used in the cookie chunking logic (cookies/index.ts:208, session-store.ts:294) which uses .length against byte limits - but after investigating now I realized that those work because cookie data is base64-encoded (pure ASCII), where .length equals byte length. 

SAML metadata is UTF-8, so that assumption doesn't hold. Will switch to Buffer.byteLength(). TY